### ISSUE TITLE
Build Script fails on Artful (17.10)

### ISSUE THREAD

**Issue by aenertia** [korcankaraokcu/PINCE#31]
Build script fails when used against gdb-8 tree that it pulls on Ubuntu 17.10 (artful)
`
make[7]: Entering directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib/import'
make[8]: Entering directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib/import'
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT dirname-lgpl.o -MD -MP -MF .deps/dirname-lgpl.Tpo -c -o dirname-lgpl.o ../.././gnulib/import/dirname-lgpl.c
mv -f .deps/dirname-lgpl.Tpo .deps/dirname-lgpl.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT basename-lgpl.o -MD -MP -MF .deps/basename-lgpl.Tpo -c -o basename-lgpl.o ../.././gnulib/import/basename-lgpl.c
mv -f .deps/basename-lgpl.Tpo .deps/basename-lgpl.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT stripslash.o -MD -MP -MF .deps/stripslash.Tpo -c -o stripslash.o ../.././gnulib/import/stripslash.c
mv -f .deps/stripslash.Tpo .deps/stripslash.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT hard-locale.o -MD -MP -MF .deps/hard-locale.Tpo -c -o hard-locale.o ../.././gnulib/import/hard-locale.c
mv -f .deps/hard-locale.Tpo .deps/hard-locale.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT localcharset.o -MD -MP -MF .deps/localcharset.Tpo -c -o localcharset.o ../.././gnulib/import/localcharset.c
mv -f .deps/localcharset.Tpo .deps/localcharset.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT malloca.o -MD -MP -MF .deps/malloca.Tpo -c -o malloca.o ../.././gnulib/import/malloca.c
mv -f .deps/malloca.Tpo .deps/malloca.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT math.o -MD -MP -MF .deps/math.Tpo -c -o math.o ../.././gnulib/import/math.c
mv -f .deps/math.Tpo .deps/math.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT strnlen1.o -MD -MP -MF .deps/strnlen1.Tpo -c -o strnlen1.o ../.././gnulib/import/strnlen1.c
mv -f .deps/strnlen1.Tpo .deps/strnlen1.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT unistd.o -MD -MP -MF .deps/unistd.Tpo -c -o unistd.o ../.././gnulib/import/unistd.c
mv -f .deps/unistd.Tpo .deps/unistd.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT wctype-h.o -MD -MP -MF .deps/wctype-h.Tpo -c -o wctype-h.o ../.././gnulib/import/wctype-h.c
mv -f .deps/wctype-h.Tpo .deps/wctype-h.Po
gcc-5 -DHAVE_CONFIG_H -I. -I../.././gnulib/import -I..     -g -O2 -MT mbrtowc.o -MD -MP -MF .deps/mbrtowc.Tpo -c -o mbrtowc.o ../.././gnulib/import/mbrtowc.c
mv -f .deps/mbrtowc.Tpo .deps/mbrtowc.Po
rm -f libgnu.a
ar cr libgnu.a dirname-lgpl.o basename-lgpl.o stripslash.o hard-locale.o localcharset.o malloca.o math.o strnlen1.o unistd.o wctype-h.o mbrtowc.o 
ranlib libgnu.a
rm -f t-charset.alias charset.alias && \
/bin/bash ../.././gnulib/import/config.charset 'x86_64-pc-linux-gnu' > t-charset.alias && \
mv t-charset.alias charset.alias
rm -f t-ref-add.sed ref-add.sed && \
sed -e '/^#/d' -e 's/@''PACKAGE''@/libgnu/g' ../.././gnulib/import/ref-add.sin > t-ref-add.sed && \
mv t-ref-add.sed ref-add.sed
rm -f t-ref-del.sed ref-del.sed && \
sed -e '/^#/d' -e 's/@''PACKAGE''@/libgnu/g' ../.././gnulib/import/ref-del.sin > t-ref-del.sed && \
mv t-ref-del.sed ref-del.sed
make[8]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib/import'
make[7]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib/import'
make[6]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib/import'
make[5]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib'
make[4]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb/build-gnulib'
make[3]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb'
g++-5 -std=gnu++11 -g -O2   -I. -I. -I./common -I./config -DLOCALEDIR="\"/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/share/locale\"" -DHAVE_CONFIG_H -I./../include/opcode -I./../opcodes/.. -I./../readline/.. -I./../zlib -I../bfd -I./../bfd -I./../include -I../libdecnumber -I./../libdecnumber  -I./gnulib/import -Ibuild-gnulib/import   -DTUI=1  -I/usr/include/python3.6m -I/usr/include/python3.6m -Wall -Wpointer-arith -Wno-unused -Wunused-value -Wunused-function -Wno-switch -Wno-char-subscripts -Wempty-body -Wunused-but-set-parameter -Wunused-but-set-variable -Wno-sign-compare -Wno-narrowing -Wformat-nonliteral  -c -o gdb.o -MT gdb.o -MMD -MP -MF .deps/gdb.Tpo gdb.c
In file included from defs.h:53:0,
                 from gdb.c:19:
ui-file.h:43:18: error: macro "putc" requires 2 arguments, but only 1 given
   int putc (int c);
                  ^
In file included from ./common/common-defs.h:78:0,
                 from defs.h:28,
                 from gdb.c:19:
./common/common-utils.h:101:12: error: ‘std::to_string’ has not been declared
 using std::to_string;
            ^
Makefile:1891: recipe for target 'gdb.o' failed
make[2]: *** [gdb.o] Error 1
make[2]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb'
Makefile:9126: recipe for target 'all-gdb' failed
make[1]: *** [all-gdb] Error 2
make[1]: Leaving directory '/vol/8tb/build/PINCE/libPINCE/gdb_pince/gdb-8.0'
Makefile:849: recipe for target 'all' failed
make: *** [all] Error 2
Failed to install GDB, restart the installation process
PINCE has been installed successfully!
Now, just run 'sh PINCE.sh' from terminal
`

**Comment by korcankaraokcu** [comment]
> error: macro "putc" requires 2 arguments, but only 1 given int putc (int c)

It seems to be a compiler-related problem. You might want to try using another compiler instead of gcc. You can do it by changing this line in `install_gdb.sh`:
`CC=gcc-5 CXX=g++-5 ./configure --prefix=$(pwd) --with-python=python3 && make && sudo make -C gdb install`
Change the `CC` part to any compiler you like. Artful (17.10) isn't included in the supported systems yet, sorry for this.

Btw, sorry for the late reply, I was in another city for a chess tournament, came back today.

**Comment by pedzed** [comment]
```
In file included from defs.h:53:0,
                 from gdb.c:19:
ui-file.h:43:18: error: macro "putc" requires 2 arguments, but only 1 given
   int putc (int c);
                  ^
In file included from ./common/common-defs.h:78:0,
                 from defs.h:28,
                 from gdb.c:19:
./common/common-utils.h:101:12: error: ‘std::to_string’ has not been declared
 using std::to_string;
            ^
Makefile:1891: recipe for target 'gdb.o' failed
make[2]: *** [gdb.o] Error 1
make[2]: Leaving directory '/home/ped/Downloads/PINCE/libPINCE/gdb_pince/gdb-8.0/gdb'
Makefile:9126: recipe for target 'all-gdb' failed
make[1]: *** [all-gdb] Error 2
make[1]: Leaving directory '/home/ped/Downloads/PINCE/libPINCE/gdb_pince/gdb-8.0'
Makefile:849: recipe for target 'all' failed
make: *** [all] Error 2
Failed to install GDB, restart the installation process
PINCE has been installed successfully!
Now, just run 'sh PINCE.sh' from terminal
```

**Comment by pedzed** [comment]
Running the program anyway gives an error as well:
```sh
$ sh PINCE.sh
```
```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
chown: cannot access '/home/ped/PINCE_USER_FILES/': No such file or directory
Detached from the process with PID:-1
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pexpect/spawnbase.py", line 150, in read_nonblocking
    s = os.read(self.child_fd, size)
OSError: [Errno 5] Input/output error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 99, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
  File "/usr/lib/python3/dist-packages/pexpect/pty_spawn.py", line 465, in read_nonblocking
    return super(spawn, self).read_nonblocking(size)
  File "/usr/lib/python3/dist-packages/pexpect/spawnbase.py", line 155, in read_nonblocking
    raise EOF('End Of File (EOF). Exception style platform.')
pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "PINCE.py", line 4309, in <module>
    window = MainForm()
  File "PINCE.py", line 300, in __init__
    GDB_Engine.init_gdb(gdb_path)
  File "/opt/PINCE/libPINCE/GDB_Engine.py", line 434, in init_gdb
    child.expect_exact("(gdb)")
  File "/usr/lib/python3/dist-packages/pexpect/spawnbase.py", line 390, in expect_exact
    return exp.expect_loop(timeout)
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 105, in expect_loop
    return self.eof(e)
  File "/usr/lib/python3/dist-packages/pexpect/expect.py", line 50, in eof
    raise EOF(msg)
pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
<pexpect.pty_spawn.spawn object at 0x7f7ecb3e9400>
command: /usr/bin/sudo
args: [b'/usr/bin/sudo', b'LC_NUMERIC=C', b'./gdb_pince/gdb-8.0/bin/gdb', b'--interpreter=mi']
buffer (last 100 chars): ''
before (last 100 chars): 'sudo: ./gdb_pince/gdb-8.0/bin/gdb: command not found\r\n'
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: 1
flag_eof: True
pid: 11289
child_fd: 11
closed: False
timeout: None
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_string:
    0: "(gdb)"
```

**Comment by goldug** [comment]
> It seems to be a compiler-related problem. You might want to try using another compiler instead of gcc. You can do it by changing this line in install_gdb.sh:
I'm going to try this since I have the same problem, but what compiler would you recommend?

**Comment by goldug** [comment]
> >  It seems to be a compiler-related problem. You might want to try using another compiler instead of gcc. You can do it by changing this line in install_gdb.sh:
> 
> I'm going to try this since I have the same problem, but what compiler would you recommend?

Eh, I tried changing from gcc-5 to plain gcc and this particular error didn't happen. However, I got another error about "makeinfo" missing and the compilation recommended me to install texinfo and gnu make (which I already have). I'll try again to see what happens.
I'm doing this all on a rather slow laptop though, so it takes ~10-20 minutes per try...

**Comment by korcankaraokcu** [comment]
@goldug Tbh, I didn't try it with a compiler other than gcc-5. So, I can't really recommend anything else by myself. Compiling process can give some errors/warnings time to time. If the gdb binary is created and the PINCE is working properly, you can ignore those errors/warnings. Good luck with your attempts, godspeed.

**Comment by korcankaraokcu** [comment]
PINCE now uses the available gcc version and gdb 8.2. This error shouldn't exist anymore. Reopen the issue if it does. About the makeinfo missing error, I get the same error as well but it doesn't affect the result so for the sake of keeping the dependencies low, I'm ignoring it at this point. PINCE will stop compiling gdb when debian compiles its gdb with python3 support anyway

**Comment by korcankaraokcu** [comment]
Another note, you can change the gdb binary in Settings->Debug
